doc/ref_arch/kubernetes/chapters/chapter02.md

```diff
@@ -304,6 +304,9 @@ The requirements in this section are to be delivered in addition to those in [se
 | `req.inf.ntw.08` | Infrastructure | Network | The Architecture **must**  provide a default network which implements the Kubernetes network model. | |
 | `req.inf.ntw.09` | Infrastructure | Network | The networking solution **must not** interfere with or cause interference to any interface or network it does not own. | |
 | `req.inf.ntw.10` | Infrastructure | Network | The Architecture **must** support cluster wide coordination of IP address assignment. | |
+| `req.inf.ntw.11` | Infrastructure |	Network | The networking solution to support multiple connection points (e.cap.004 - please see [Section 2.2.1](#2.2.1)) must conform to the Kubernetes Networking CRD de-facto standard ([K8S Network Plumbing Working Group](https://github.com/k8snetworkplumbingwg/community); [CRD](https://docs.google.com/document/d/1Ny03h6IDVy_e_vmElOqR7UdTPAG_RNydhVE1Kx54kFQ/edit)). | |
```

Comment:
This spec unfortunately never made it to be a Kubernetes API, therefore it _not_ supported by Kubernetes so I suggest to remove the refernce to it.

Reply:
@CsatariGergely I am very confused. I thought the purpose of a CRD was to provide API extensibility and flexibility. Don't the CRDs extend the Kubernetes APIs?

Reply:
Yes, CRD-s are to extend the Kubernetes API and this is exactly what my point is. The CRD-s specified in the linked document were never implemented as an in tree API in Kubernetes.

Reply:
@CsatariGergely Not sure what you are trying to state here:
- as you know, a CRD doesn't affect the default Kubernetes installation so the default Kubernetes API tree will be unaffected
- when a CRD is incorporated in a custom installation (using kubectl apply), the API tree does get affected as it then includes the new API endpoints

Reply:
@pgoyal01 could you commit my suggestion above, and then we can see if there are any objections to that?

Reply:
My point was that the CRD-s defined by the network plumbing WG are not part of upstream Kubernetes there fore I would not say that it is a de-facto standard. The defined API does not provide an abstract way to represent network needs, but provides direct access to the underlying CNI-s configuration (You can imagine it as the SR-IOV or metaplugin CRD-s).
I'm okay with the wording to use CRD-s in general.

Reply:
@CsatariGergely Isn't a CRD by definition not part of upstream kubernetes?

Reply:
OK here's another thought - does this requirement limit us in terms of the possible solutions available to us (e.g. NSM, which could provide the interface via the Device Plugin API)?  I don't think "metaplugin" is the only possible solution to multiple connection points?

Reply:
@CsatariGergely are you ok with the updated wording, or still a problem?

Reply:
@tomkivlin  yes, the current wording limits us to Multus as the only Implementation of the Network Plumbing WG-s spec.  I'm not ok with the current wording as it still requires compliancy with the Network Plumbing WG-s CRD spec what is designed in a way that the users can bypass Kubernetes security framework.
Without `req.inf.ntw.11` I'm okey with the changes.

Reply:
@pgoyal01 could you either address Gergely's concern by removing .011 so we can merge for Baraque, or remove the "Baraque" label and M3 milestone so we deal with it in the next release.

Reply:
@tomkivlin It is a CRD spec. The wording nowhere limits it to the MULTUS implementation.  Shouldn't the purpose of our specification be to rely on other specs. Maybe others should be asked to chime in on this particular requirement.

For now I am removing the Baraque and M3 label.

Reply:
First up, yes - we should rely on other specs but only where they are either widely adopted or a formal standard (in my opinion).  For me, the Kubernetes networking space is still settling and I'm not sure restricting the networking solutions that could be used is useful as it'll stifle innovation in a fast-moving area.

Re. the wording, the working group refers to Multus as "the" Reference Implementation of the CRD spec ([here](https://github.com/k8snetworkplumbingwg/community#working-items)).  So yes I agree it doesn't limit an operator to use Multus, but I suspect various valid solutions such as e.g. NSM would not be useable as a network solution, even if it met all other requirements.  As you say, it would be useful for others to chime in.

Reply:
I would like to reach consensus on this.  For me, a spec from the working group set up to create Multus isn't a good enough source or standard.  If it were to come from SIG-Network then it's a different story.  As per the Tech Steering call just now, we know there is a gap when it comes to the standard management of multiple interfaces.  My suggestion would be to create an issue that gets us to document this gap in ch6, and we remove the CRD part of this change - the rest I'm fine with.  @pgoyal01 @CsatariGergely @walterkozlowski @gkunz @petorre @collivier @ulikleber  do you agree?

Reply:
@pgoyal01 correct, I already have an open issue for that: https://github.com/cntt-n/CNTT/issues/1911

Reply:
@tomkivlin Can you remove the hold and also let the associated PR #1912 get approved now so that the Baraque release is consistent.

Reply:
@tomkivlin  req.inf.ntw.11 commented out: "The networking solution to support multiple connection points (e.cap.004 - please see [Section 2.2.1](#2.2.1)) must conform to the [Multi-Network CRD Specification from the Kubernetes Network Plumbing Working Group](https://github.com/k8snetworkplumbingwg/multi-net-spec)."

Reply:
@tomkivlin yes, I kind of agree with this conclusion for the time being. I wasn't able to attend today's TSC, but the biggest issue for me is that each implementation (Multus, DANM, NSM) currently brings its own API (CRD) as all of those follow (slightly) different designs. Can we build a complex system without standardized APIs? No. Are the currently existing solutions good enough to solve the Kubernetes telco networking problem fully? Probably not. So, this is a good example of where CNTT's approach of building specs from existing solutions is challenged and we certainly need to step up our efforts of "influencing" the technical development of upstream communities.